compiler/rustc_trait_selection/src/traits/const_evaluatable.rs

```diff
@@ -668,8 +668,19 @@ where
         ct: AbstractConst<'tcx>,
         f: &mut dyn FnMut(AbstractConst<'tcx>) -> ControlFlow<R>,
     ) -> ControlFlow<R> {
-        f(ct)?;
         let root = ct.root(tcx);
+        if let Node::Leaf(leaf) = root {
+            if let ty::ConstKind::Unevaluated(uv) = leaf.kind() {
+                if let Ok(Some(ac)) = AbstractConst::new(tcx, uv.shrink()) {
+                    return recurse(tcx, ac, f);
+                } else if uv.substs.has_param_types_or_consts()
+                    && tcx.def_kind(uv.def.did) == DefKind::AnonConst
+                {
+                    return ControlFlow::CONTINUE;
```

Comment:
do you also mean the identity subst component?

Reply:
yeah, need that because we don't have a param env in `AbstractConst`, so not doing that would cause ICE :thinking: though maybe we should require one in `AbstractConst::new`. Then the identity substs stuff isn't required anymore.